lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -1161,39 +1167,46 @@ bool GrandCentralPass::traverseField(Attribute field, IntegerAttr id,
         auto builder =
             OpBuilder::atBlockEnd(companionIDMap.lookup(id).mapping.getBody());
 
-        // The enclosing module is either the module where the leaf lives or its
-        // the root of the NLA.  After computing this, the enclosing module
-        // should be singly-instantiated.
-        FModuleLike enclosing;
-        if (!nla)
-          enclosing = getEnclosingModule(leafValue, sym);
-        else
-          enclosing = getSymbolTable().lookup<FModuleOp>(nla.root());
+        // Populate a hierarchical path to the leaf.  For an NLA this is just
+        // the namepath of the associated hierarchical path.  For a local
+        // annotation, this is computed from the instance path.
+        SmallVector<Attribute> fullLeafPath;
+        if (nla) {
+          fullLeafPath.append(nla.namepath().begin(), nla.namepath().end());
+        } else {
+          FModuleLike enclosing = getEnclosingModule(leafValue, sym);
+          auto enclosingPaths = instancePaths->getAbsolutePaths(enclosing);
+          assert(enclosingPaths.size() == 1 &&
+                 "Unable to handle multiply instantiated companions");
+          if (enclosingPaths.size() != 1)
+            return false;
+          StringAttr root =
+              instancePaths->instanceGraph.getTopLevelModule().moduleNameAttr();
```

Comment:
Seems like this handles the case when `DUT` is the top level module, which might not be possible. but it makes sense!